SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -246,6 +250,51 @@ - (void)didEnterStateConnected {
 }
 
 - (void)didEnterStateRegistered {
+    NSArray<SDLLanguage> *supportedLanguages = self.configuration.lifecycleConfig.languagesSupported;
+    SDLLanguage desiredLanguage = self.configuration.lifecycleConfig.language;
+    SDLLanguage actualLanguage = self.registerResponse.language;
+    BOOL delegateCanUpdateLifecycle = [self.delegate respondsToSelector:@selector(managerWillUpdateLifecycleToLanguage:)];
+    
+    // language mismatch? but actual language is a supported language? and delegate has implemented method?
+    if (actualLanguage != desiredLanguage && [supportedLanguages containsObject:actualLanguage] && delegateCanUpdateLifecycle) {
+        [self.lifecycleStateMachine transitionToState:SDLLifecycleStateUpdatingConfiguration];
+    } else {
+        [self.lifecycleStateMachine transitionToState:SDLLifecycleStateSettingUpManagers];
+    }
+}
+
+- (void)didEnterStateUpdatingConfiguration {
+    // we can expect that the delegate has implemented the update method and the actual language is a supported language
+    SDLLanguage actualLanguage = self.registerResponse.language;
+
+    SDLLifecycleConfigurationUpdate *configUpdate = [self.delegate managerWillUpdateLifecycleToLanguage:actualLanguage];
+    
+    if (configUpdate) {
+        self.configuration.lifecycleConfig.language = actualLanguage;
+        if (configUpdate.appName) {
+            self.configuration.lifecycleConfig.appName = configUpdate.appName;
+        }
+        if (configUpdate.shortAppName) {
+            self.configuration.lifecycleConfig.shortAppName = configUpdate.shortAppName;
+        }
+        if (configUpdate.ttsName) {
+            self.configuration.lifecycleConfig.ttsName = configUpdate.ttsName;
+        }
+        if (configUpdate.voiceRecognitionCommandNames) {
+            self.configuration.lifecycleConfig.voiceRecognitionCommandNames = configUpdate.voiceRecognitionCommandNames;
+        }
+        
+        SDLChangeRegistration *changeRegistration = [[SDLChangeRegistration alloc] init];
```

Comment:
Why not use the more comprehensive initializer here?

Reply:
In this case it's necessary to set all the parameters. Using the initializer would either cause a huge single line or multiple lines which are indented far to the right.

Reply:
The huge single-line is fine because of wrapping. That's the preferred way to do it within the project.

Reply:
You're asking for the exact same result which makes it more difficult to read the code. I would leave it as is

Reply:
This is a blocker before it can be merged. Please make the requested change.